File: command_pattern/python/2-bank-state-focus/account.py
from dataclasses import dataclass


@dataclass
class Account:
    """
    The account object will be our receiver object.

    It will be the object that knows how to execute the action that the command object will encapsulate.
    """

    name: str
    number: str
    balance: float = 0.0
    
    def deposit(self, amount: float) -> None:
        self.balance += amount
        
    def withdraw(self, amount: float) -> None:
        if amount > self.balance:
            raise ValueError('Insufficient funds')
        
        self.balance -= amount

File: command_pattern/java/2-bank-state-focus/receiver/Account.java
package receiver;


public class Account {

    String name;
    String number;
    private float balance;

    public Account(String name, String number) {
        this.name = name;
        this.number = number;
        this.balance = 0;
    }

    public void deposit(float amount) {
        this.balance += amount;
    }

    public void withdraw(float amount) {

        if (amount > this.balance) {
            throw new IllegalArgumentException("Insufficient funds");
        }
        this.balance -= amount;
    }
}